### ISSUE TITLE
(Feature Request): Option for content creators to submit their private dislike counter to the database

### ISSUE THREAD

**Issue by teeuwen** [Anarios/return-youtube-dislike#56]
### Extension or Userscript?

Both

### Request or suggest a new feature!

Many content creators have already shown their disapproval of the removal of the public dislike counters on YouTube. However, the data is and will remain visible to creators themselves.

Therefore I thought it'd be a good idea to somehow integrate these private statistics into the public database.

### Ways to implement this!

From December 13th onwards, return-youtube-dislike will be using their own database (amongst other techniques) to display the like/dislike ratio.

As YouTube stated in an email regarding the removal of the public discount field; the statistic will still be visible to an authenticated user who owns the video itself:

> The dislike_count field within the statistics part of the video resource will be omitted on calls to the video.list endpoint except in cases in which the request is being authenticated as a user (such as the creator or the agent user) who owns the video that is being requested.

**My proposal is as follows:**
A content creator would, via a locally running client application that interfaces with the YouTube API, automatically submit their results to the return-youtube-dislike database periodically. This application would obviously be open-source and not submit any other non-relevant statistics to the database to protect him/her.

The statistic would subsequently overwrite/merge with the database or other methods of acquiring the like/dislike ratio as long as the data is not severely outdated. This is just an example, I haven't yet thought about how the data would be processed.

I'm curious if something like this would be picked up by creators.
I think the idea is sufficiently simple to implement that it is worth putting time into this.

### Can you work on this?

- [X] Yes
- [ ] No

**Comment by lishangqiu** [comment]
It is possible that the creator spoofs the dislikes? Ultimately it all comes down to a POST request and some sort of dislike number in the body message, and people can pretty easily spoof that.

**Comment by teeuwen** [comment]
A very good point you bring up there.

Running the application locally seemed like a good idea initially (with the idea that people always have best intentions). Preventing spoofing completely seems basically impossible if the program is running client-side.

Which brings us to reconsidering running the application on a server which interfaces with the YT API.
Fortunately, [scopes](https://developers.google.com/youtube/v3/guides/auth/server-side-web-apps#identify-access-scopes) are included in the API via the `readonly` scope.

This raises some new privacy concerns however as I think this also allows for viewing private playlists, videos, statistics, watch history, etc. This seems like quite a turn off to me.

**Comment by Anarios** [comment]
Yep, this was in plans since conception of the extension. We could show scores of creators who consented to share data in some special way (green highlight that it's verified, or something similar)

**Comment by sy-b** [comment]
> It is possible that the creator spoofs the dislikes? Ultimately it all comes down to a POST request and some sort of dislike number in the body message, and people can pretty easily spoof that.

We may not be able to stop that but we can certainly make it harder.

**Comment by sy-b** [comment]
@Anarios 
I think we must implement this well before 13th. It'll have no meaning if creators & users get acclimatized to hidden dislikes.

**Comment by sy-b** [comment]
We'll may need some review from YT creators to improve it. especially from "non-tech savvy creators".

In my opinion, it must be just one click for them (and an option to do it automatically).

**Comment by folknor** [comment]
> In my opinion, it must be just one click for them (and an option to do it automatically).

There's no need for an option. They opted in when they installed the extension. It should always be automatic. KISS!

**Comment by sy-b** [comment]
Can we have this before 2021-11-15T00:40:50Z 
(and release it on RYD's birthday)
